config/_templates/dataset/carla_garage/town12.yaml

```diff
@@ -0,0 +1,1996 @@
+#@  'YieldToEmergencyVehicle/Town12_Rep0_2465_1_route0_11_09_11_02_39',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_2489_0_route0_11_08_12_21_04',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_2536_0_route0_11_08_13_56_54',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_2536_1_route0_11_09_12_15_45',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_2958_0_route0_11_09_09_09_28',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_2958_1_route0_11_09_12_23_56',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_2983_0_route0_11_09_11_54_29',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_3998_0_route0_11_09_08_51_31',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_3998_1_route0_11_09_01_40_49',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4019_0_route0_11_08_17_36_09',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4019_1_route0_11_08_15_18_26',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4065_0_route0_11_08_00_44_13',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4065_1_route0_11_08_02_37_34',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4116_0_route0_11_08_01_51_54',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4116_1_route0_11_09_11_20_43',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4123_0_route0_11_08_19_43_31',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4329_0_route0_11_09_01_08_23',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4329_1_route0_11_08_23_44_22',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4477_0_route0_11_08_00_46_21',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4477_1_route0_11_08_03_32_21',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4483_0_route0_11_08_07_19_24',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4483_1_route0_11_08_12_48_13',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4485_0_route0_11_09_02_17_39',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4485_1_route0_11_08_03_40_13',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4511_0_route0_11_07_23_55_40',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4516_0_route0_11_08_01_51_30',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4516_1_route0_11_09_04_21_09',
+#@  'YieldToEmergencyVehicle/Town12_Rep0_4524_0_route0_11_08_00_49_44',
+#@ ]
+#@ cameras = [
+#@     'rgb',
+#@ ]
+---
+_target_: rbyte.Dataset
+_recursive_: false
+_convert_: all
+sources:
+  #@ for input_id in drives:
+  (@=input_id@):
+    #@ for source_id in cameras:
+    (@=source_id@):
+      index_column: meta._idx_
+      source:
+        _target_: rbyte.io.PathTensorSource
+        path: '${paths.data}/(@=input_id@)/(@=source_id@)/{:04d}.jpg'
+        decoder:
+          _target_: simplejpeg.decode_jpeg
+          _partial_: true
+          colorspace: rgb
+          fastdct: true
+          fastupsample: true
+
+    #@ end
+  #@ end
+
+samples:
+  inputs:
+    #@ for input_id in drives:
+    (@=input_id@):
+      measurements_path: ${paths.data}/(@=input_id@)/measurements.json
+      waypoints_path: ${paths.data}/(@=input_id@)/waypoints.json
+    #@ end
+
+  executor:
+    _target_: concurrent.futures.ThreadPoolExecutor
+
+  storage: dict
```

Comment:
not sure if dataset init is a bottleneck at all in this case but could probably be sped up, see https://github.com/yaak-ai/rmind/blob/main/config/_templates/dataset/yaak/train.yaml#L723-L731